app/workers/scheduled_publishing_worker.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+class ScheduledPublishingWorker
+  include Sidekiq::Worker
+  sidekiq_options retry: 10
+  sidekiq_retry_in { 30 }
+
+  def perform(id)
+    Edition.transaction do
+      edition = Edition.lock.find(id)
+
+      unless edition.scheduled?
+        raise "Cannot schedule an edition that is not in a scheduled state"
+      end
+
+      if scheduled_publishing_datetime_in_the_past?(edition)
+        raise "Cannot schedule an edition whose scheduled publishing datetime is in the past"
+      end
+
+      user = edition.status.created_by
+      reviewed = edition.status.details&.reviewed
```

Comment:
I think we can avoid the safe navigation operator here as we always expect them to have scheduled details don't we?